### DISCUSSION TITLE
I'm still having audio issues in Linux

### DISCUSSION THREAD

**Discussion by mitchdotnet** [AlexandreRouma/SDRPlusPlus#1328] [General]
Alright, so I'm having a fun time getting SDR++ working correctly on my Thinkpad under Arch Linux. Initially I had issues with the default rtaudio sink so I recompiled rtaudio as suggested in https://github.com/AlexandreRouma/SDRPlusPlus/discussions/112. At first I thought this solved my problems but it's still not quite working right. When launching SDR++ and starting the source the FFT and waterfall both show activity as expected but there is no audio output. If I use pavucontrol to switch the profiles around from "Play HiFi Music" which was apparently the default to "Pro Audio" then immediately I get working sound. I can even then switch it back to "Play HiFi Music" and it continues to work, at least until I close SDR++ and reopen it in which case I have to do this all over again. I also can't just leave it on "Pro Audio" as this causes issues with pretty much every other application. I'm pretty much in the same boat as I was before rebuilding rtaudio where I'd have to switch the sample rate of the audio sink back and forth to get the audio to stop skipping only now this is kind of more annoying since I have to open another application to get the audio to work instead of just being able to fix it right from within SDR++.

I think the root cause of all of this is some weird issue between my sound hardware, the sof-firmware driver, pipwire, and rtaudio. I don't have audio problems in any other app, and SDR++ works just fine on my desktop which is also running Arch but doesn't require the sof driver. On the desktop I did have to build SDR++ to use the portaudio sink but that made things even worse when I tried it on my laptop. It's also possible that I just have things severely misconfigured; my knowledge of the inner workings of Linux audio are slim at best, but like I said this is the only application I have audio issues with. Basically I'm hoping someone here is more familiar with this stuff and/or has gotten SDR++ working with a sound card that uses the sof firmware and can give me some pointers on where to start looking.

**Comment by 4mosoT** [comment]
I was having problem too. I installed all the related sounds packages ALSA/Pulseaudio with all the extras, alsa-plugins, etc.. 
Then install this AUR package: https://aur.archlinux.org/packages/rtaudio5-exp
Finally this one: https://aur.archlinux.org/packages/sdrpp-git
All is now fine to me.
